crates/isograph_schema/src/validate_entrypoint.rs

```diff
@@ -1,34 +1,76 @@
-use std::collections::HashMap;
+use std::collections::{hash_map::Entry, HashMap};
+use std::fmt;
 
 use common_lang_types::{
     IsoLiteralText, IsographObjectTypeName, Location, ServerScalarSelectableName, TextSource,
     UnvalidatedTypeName, WithLocation, WithSpan,
 };
 use isograph_lang_types::{
-    ClientScalarSelectableId, DefinitionLocation, EntrypointDeclaration, SelectionType,
-    ServerEntityId, ServerObjectEntityId,
+    ClientScalarSelectableId, DefinitionLocation, EntrypointDeclaration, EntrypointDirectiveSet,
+    SelectionType, ServerEntityId, ServerObjectEntityId,
 };
 
 use thiserror::Error;
 
 use crate::{NetworkProtocol, Schema};
 
+#[derive(Debug)]
+pub struct LoadTypedIsoLiteralText {
+    pub iso_literal_text: IsoLiteralText,
+    pub is_lazy_loaded: bool,
+}
+
+impl LoadTypedIsoLiteralText {
+    pub fn new(entrypoint_declaration: WithSpan<EntrypointDeclaration>) -> Self {
```

Comment:
I think we can just inline this, there's no need to have a separate function here IMO